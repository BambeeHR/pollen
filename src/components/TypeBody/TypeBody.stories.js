import Typography from '../../constants/Typography';
import TypeBody from './TypeBody.vue';
import TypeOverline from '../TypeOverline/TypeOverline.vue';

const { Body, Weights } = Typography;

export default {
  title: 'Typography/TypeBody',
  component: TypeBody,
};

export const Gallery = () => ({
  components: { TypeBody, TypeOverline },
  data() {
    return {
      variants: Object.values(Body),
      weights: Object.values(Weights),
    };
  },
  template: `
    <div>
      <div v-for="variant in variants" :key="variant" :style="{ marginBottom: '48px' }">
        <TypeOverline tag="h1">{{ variant }}</TypeOverline>
        <TypeBody 
          v-for="weight in weights"
          :key="variant + '-' + weight"
          :variant="variant"
          :weight="weight"
          tag="div"
        >
          The quick brown fox jumps over the lazy dog.
        </TypeBody>
      </div>
    </div>
  `,
});
