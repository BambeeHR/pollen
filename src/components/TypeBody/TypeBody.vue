<template functional>
  <component
    :is="props.tag"
    :class="[
      `type-body-${props.variant}`,
      `type-body-${props.variant}__${props.weight}`,
      data.class,
      data.staticClass,
    ]"
    class="type-body"
    v-bind="$options.getAttributes({ data, parent })"
    v-on="listeners"
  >
    <slot />
  </component>
</template>

<script>
import Typography from '../../constants/Typography';
import getAttributes from '../../util/getAttributes';

const { Body, Weights } = Typography;

/**
 * A styled block of text.
 */
export default {
  props: {
    /** The html tag to use for this component. */
    tag: {
      type: String,
      default: 'span',
    },
    /** One of `legal`, `caption`, `small`, `text`, `medium`, or `large`. */
    variant: {
      type: String,
      default: Typography.Body.TEXT,
      validator: (value) => Object.values(Body).includes(value),
    },
    /** One of `regular`, `medium`, or `strong`. */
    weight: {
      type: String,
      default: Typography.Weights.REGULAR,
      validator: (value) => Object.values(Weights).includes(value),
    },
  },
  getAttributes,
};
</script>

<style>
/* Intentionally left unscoped to provide global helper styles. */
@each $variant in legal, caption, small, text, medium, large {
  .type-body-$(variant) {
    @apply font-body-$(variant);
  }

  .type-body-$(variant)__regular {
    @apply font-normal;
  }

  .type-body-$(variant)__medium {
    @apply font-medium;
  }

  .type-body-$(variant)__strong {
    @apply font-semibold;
  }
}
</style>
