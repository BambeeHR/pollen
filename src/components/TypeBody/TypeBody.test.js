import { mount } from '@vue/test-utils';
import TypeBody from './TypeBody.vue';

describe('TypeBody', () => {
  test('renders correctly', () => {
    const wrapper = mount(TypeBody, {
      slots: {
        default: 'Hello world!',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
