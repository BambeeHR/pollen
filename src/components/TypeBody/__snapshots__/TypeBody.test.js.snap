// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeBody renders correctly 1`] = `
<span
  class="type-body type-body-text type-body-text--regular"
>
  Hello world!
</span>
`;
